src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -1,12 +1,66 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import lombok.RequiredArgsConstructor;
+import org.hl7.fhir.dstu3.model.Annotation;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
 import org.hl7.fhir.dstu3.model.ProcedureRequest;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.FreeTextSegment;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ReportStatusCode;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.PatientClinicalInfo;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
 public class ProcedureRequestMapper {
-    public ProcedureRequest mapToProcedureRequest(final Message message) {
-        return new ProcedureRequest();
+    private final UUIDGenerator uuidGenerator;
+    private static final Map<ReportStatusCode, ProcedureRequest.ProcedureRequestStatus> STATUS_CODE_MAPPING = Map.of(
+        ReportStatusCode.UNSPECIFIED, ProcedureRequest.ProcedureRequestStatus.UNKNOWN
+    );
+
+    public Optional<ProcedureRequest> mapToProcedureRequest(final Message message) {
+        return message.getServiceReportDetails().getSubject().getClinicalInfo()
+            .map(this::mapPatientClinicalInfo);
+    }
+
+    private ProcedureRequest mapPatientClinicalInfo(final PatientClinicalInfo patientClinicalInfo) {
+        final ProcedureRequest procedureRequest = new ProcedureRequest();
+
+        mapFreeText(patientClinicalInfo, procedureRequest);
+        mapStatus(patientClinicalInfo, procedureRequest);
+        procedureRequest.setIntent(ProcedureRequest.ProcedureRequestIntent.NULL);
+        procedureRequest.setCode(new CodeableConcept().setText("unknown"));
+        procedureRequest.setId(uuidGenerator.generateUUID());
+
+        return procedureRequest;
+    }
+
+    private void mapFreeText(final PatientClinicalInfo patientClinicalInfo, final ProcedureRequest procedureRequest) {
+        final List<Annotation> annotations = patientClinicalInfo.getFreeTexts().stream()
+            .map(FreeTextSegment::getTexts)
+            .map(texts -> String.join(" ", texts))
+            .map(MappingUtils::unescape)
+            .map(text -> new Annotation().setText(text))
+            .collect(Collectors.toList());
+
+        if (annotations.isEmpty()) {
+            throw new FhirValidationException("Unable to map message. "
+                + "The FreeText segment is mandatory in Clinical Information");
+        }
```

Comment:
You could move this to the top of the method, so the exception is thrown early if there's no free text (rather than trying to do all the mapping, and then throw an error)

```
List<FreeTextSegment> patientClinicalInfoFreeTexts = patientClinicalInfo.getFreeTexts();

if (patientClinicalInfoFreeTexts.isEmpty()) {
    throw new FhirValidationException("Unable to map message. "
        + "The FreeText segment is mandatory in Clinical Information");
}

final List<Annotation> annotations = patientClinicalInfoFreeTexts.stream()
    .map(FreeTextSegment::getTexts)
    .map(texts -> String.join(" ", texts))
    .map(MappingUtils::unescape)
    .map(text -> new Annotation().setText(text))
    .collect(Collectors.toList());

procedureRequest.setNote(annotations);
```

Reply:
I don't think `patientClinicalInfoFreeTexts` can be empty? It's validated in the `FreeTextSegment`

Reply:
@matthewdoreilly Thanks! Fixed in ae28749855ad4178a2717ba63b29da9bb9d76924

@n-r-b I think it can because it seems that it can return the empty list. I gave a more detailed answer [here](https://github.com/nhsconnect/integration-adaptor-lab-results/pull/65#discussion_r588398346). Tell me what do you think about that, please :)